crypto/zkp/primitives/commitments/damgard-fujisaki_positive.go

```diff
@@ -110,62 +110,68 @@ func getCommitRandoms(r *big.Int, numOfRoots int) []*big.Int {
 
 func (p *DFCommitmentPositiveProver) GetProofRandomData() []*big.Int {
 	proofRandomData := make([]*big.Int, len(p.squareProvers)*2)
-	for index, squareProver := range p.squareProvers {
+	for i, squareProver := range p.squareProvers {
 		proofRandomData1, proofRandomData2 := squareProver.GetProofRandomData()
-		proofRandomData[2*index] = proofRandomData1
-		proofRandomData[2*index+1] = proofRandomData2
+		proofRandomData[2*i] = proofRandomData1
+		proofRandomData[2*i+1] = proofRandomData2
 	}
 	return proofRandomData
 }
 
 func (p *DFCommitmentPositiveProver) GetProofData(challenges []*big.Int) []*big.Int {
 	proofData := make([]*big.Int, len(p.squareProvers)*3)
-	for index, squareProver := range p.squareProvers {
-		s1, s21, s22 := squareProver.GetProofData(challenges[index])
-		proofData[3*index] = s1
-		proofData[3*index+1] = s21
-		proofData[3*index+2] = s22
+	for i, squareProver := range p.squareProvers {
+		s1, s21, s22 := squareProver.GetProofData(challenges[i])
+		proofData[3*i] = s1
+		proofData[3*i+1] = s21
+		proofData[3*i+2] = s22
 	}
 	return proofData
 }
 
+// GetVerifierInitializationData returns data that are needed by DFCommitmentPositiveVerifier
+// and are known only after the initialization of DFCommitmentPositiveProver.
+func (p *DFCommitmentPositiveProver) GetVerifierInitializationData() ([]*big.Int, []*big.Int) {
```

Comment:
If we have this function in order for the caller to access prover's commitments, then we can safely make `struct DFCommitmentPositiveProver`'s fields `SmallCommitments` and `BigCommitments` unexported (e.g. `smallCommitments` and `bigCommitments`. Please rename them.